### PR TITLE
EVA-978 - Deal with variants that have IUPAC codes

### PR DESCRIPTION
**Pull request by sundarvenkata-EBI**

### CODE REVIEW

File: dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java [hunk 1]
```diff
@@ -36,16 +54,88 @@ public SubSnpCoreFieldsToVariantProcessor(int dbsnpBuild) {
 
     @Override
     public Variant process(SubSnpCoreFields subSnpCoreFields) throws Exception {
+        if (areGenotypesInvalid(subSnpCoreFields)) {
+            logger.debug("Filter out variant because of invalid genotypes: {}, {}", subSnpCoreFields,
```
[the remark this answers is not in the document]
Comment (jmmut) [reply dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java]: null or empty genotypes are also invalid and enough to filter the variant out.

File: dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java [hunk 2]
```diff
@@ -36,16 +54,88 @@ public SubSnpCoreFieldsToVariantProcessor(int dbsnpBuild) {
...
+                                                             String[] secondaryAlternates,
+                                                             boolean alleleOrientation) {
+        return Arrays.stream(genotypesString.split(",", -1))
+                     .map(genotypeString -> getForwardOrientedGenotypeCodes(alleleOrientation, genotypeString, ref, alt,
+                                                                            secondaryAlternates))
+                     .map(genotypeCodesString -> getGenotypeMapFromString(genotypeCodesString))
+                     .collect(Collectors.toList());
+    }
+
+    private String getForwardOrientedGenotypeCodes(boolean forward, String genotypeString, String ref,
```
[the remark this answers is not in the document]
Comment (cyenyxe) [reply dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java]: Agreed offline to keep this one as `genotypeCodes`.

File: dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java [hunk 3]
```diff
@@ -36,16 +54,88 @@ public SubSnpCoreFieldsToVariantProcessor(int dbsnpBuild) {
...
+        return getGenotypesFromString(subSnpCoreFields.getRawGenotypesString(),
+                                      subSnpCoreFields.getReferenceInForwardStrand(),
+                                      subSnpCoreFields.getAlternateInForwardStrand(),
+                                      subSnpCoreFields.getSecondaryAlternatesInForwardStrand(),
+                                      subSnpCoreFields.getAlleleOrientation());
+    }
+
+    private List<Map<String, String>> getGenotypesFromString(String genotypesString, String ref, String alt,
+                                                             String[] secondaryAlternates,
+                                                             boolean alleleOrientation) {
```
[the remark this answers is not in the document]
Comment (jmmut) [reply dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java]: isForward is more explanatory IMO

File: dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmmut) [reply dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SubSnpCoreFieldsToVariantProcessor.java]: changed to start exactly like the other filter messages

File: dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/configuration/ImportVariantsStepConfigurationTest.java [hunk 4]
```diff
@@ -52,7 +59,7 @@
 @RunWith(SpringRunner.class)
 @TestPropertySource({"classpath:application.properties"})
 @ContextConfiguration(classes = {ImportEvaSubmittedVariantsJobConfiguration.class, MongoTestConfiguration.class,
```
Comment (pabarcgar) [inline dbsnp-importer/src/test/java/uk/ac/ebi/eva/dbsnpimporter/configuration/ImportVariantsStepConfigurationTest.java]: Change from ImportEvaSubmittedVariantsJobConfiguration to ImportsVariantsJobConfiguration

**Pull request merged**